File: spring/src/main/java/com/lina/spring/models/MeteoActuelle.java
package com.lina.spring.models;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class MeteoActuelle {
  public String meteo;
  public String meteoIconUrl;
  public String meteoIconBase64;
  public String couvNuageuse;
  public String temperature;
  public String tempRessentis;
  public String errorMessage;

  public MeteoActuelle(String meteo, String meteoIconUrl, String meteoIconBase64,
                       String couvNuageuse, String temperature, String tempRessentis) {
    this.meteo = meteo;
    this.meteoIconUrl = meteoIconUrl;
    this.meteoIconBase64 = meteoIconBase64;
    this.couvNuageuse = couvNuageuse;
    this.temperature = temperature;
    this.tempRessentis = tempRessentis;
    this.errorMessage = "";
  }

  public MeteoActuelle(String errorMessage) {
    this.meteo = "";
    this.meteoIconUrl = "";
    this.meteoIconBase64 = "";
    this.couvNuageuse = "";
    this.temperature = "";
    this.tempRessentis = "";
    this.errorMessage = errorMessage;
  }
}

File: src/app/model/MeteoActuelle.ts
export class MeteoActuelle {
  meteo: string;
  meteoIconURL: string;
  meteoIconBase64: string;
  couvNuageuse: string;
  temperature: string;
  tempRessentis: string;

  constructor(meteo?: string, meteoIconURL?: string, meteoIconBase64?: string, couvNuageuse?: string, temperature?: string, tempRessentis?: string) {
    this.meteo = (meteo) ? meteo : '';
    this.meteoIconURL = (meteoIconURL) ? meteoIconURL : '';
    this.meteoIconBase64 = (meteoIconBase64) ? meteoIconBase64 : '';
    this.couvNuageuse = (couvNuageuse) ? couvNuageuse : '';
    this.temperature = (temperature) ? temperature : '';
    this.tempRessentis = (tempRessentis) ? tempRessentis : '';
  }
}
